packages/core/src/new-api/internal/wiper.ts

```diff
@@ -0,0 +1,52 @@
+import { IgnitionError } from "../../errors";
+import { Journal, WipeMessage } from "../types/journal";
+
+import { executionStateReducer } from "./execution/executionStateReducer";
+import { ExecutionStateMap } from "./types/execution-state";
+
+export class Wiper {
```

Comment:
This is matching the pattern of an internal Deployer and a public `deploy` function.